### PR TITLE
Resolve task

### PR DESCRIPTION
**Pull request by YaroslavKust**

### CODE REVIEW

File: Module 4/reddit_posts/src/actions/postListActionCreators.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const markAsRead = (postId) =>{
+    return{
+        type: actionTypes.MARK_AS_READ,
+        id: postId
+    }
+}
+
+const loadAdditional = (after) =>{
+    console.log("load action");
```
Comment (mol4) [inline Module 4/reddit_posts/src/actions/postListActionCreators.js:26]: Needs to delete all console logs in prod version

File: Module 4/reddit_posts/src/components/PostDetails.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { connect } from "react-redux";
+import { relativeTime } from "../utils/relativeTime";
+
+const PostDetails = ({item}) =>{
+    let data = item.post;
+    console.log(data);
+    if(data != undefined)
```
Comment (mol4) [inline Module 4/reddit_posts/src/components/PostDetails.js:7]: The best option is 
`if (data) {}`

File: Module 4/reddit_posts/src/components/PostDetails.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                <p><label>author: </label>{data.author}</p>
+                <p>{relativeTime(new Date(),data.date * 1000)}</p>
+                <p><a target="_blank" href={data.thumbnail}><img src={data.thumbnail} width={300} height={300}></img></a></p>
+                <p><label>comments: </label>{data.comments}</p>
+            </div>
+        )
+    return null;
+}
+
+const mapStateToProps = state =>(
```
Comment (mol4) [inline Module 4/reddit_posts/src/components/PostDetails.js:20]: in function components needs to use the hook - useSelector

File: Module 4/reddit_posts/src/components/PostList.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            }>Remove all</button>
+            <TransitionGroup>
+                {postsJsx}
+            </TransitionGroup>
+            <button onClick={()=>loadAdditional(items.after)}>Load</button>
+        </div>
+    )
+}
+
+const mapStateToProps = state =>({
```
Comment (mol4) [inline Module 4/reddit_posts/src/components/PostList.js:46]: useSelector

**Review by mol4 — CHANGES_REQUESTED** [review]

**Comment by YaroslavKust** [comment]
fixed

**Pull request merged**